cylc/rose/entry_points.py

```diff
@@ -0,0 +1,208 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Cylc support for reading and interpreting ``rose-suite.conf`` workflow
+configuration files.
+
+Top level module providing entry point functions.
+"""
+
+import os
+
+from pathlib import Path
+
+from metomi.rose.config import ConfigLoader, ConfigDumper
+from cylc.rose.utilities import (
+    get_rose_vars_from_config_node,
+    rose_config_exists,
+    rose_config_tree_loader,
+    merge_rose_cylc_suite_install_conf,
+    get_cli_opts_node,
+    add_cylc_install_to_rose_conf_node_opts,
+)
+
+
+class MultipleTemplatingEnginesError(Exception):
+    ...
+
+
+def get_rose_vars(dir_=None, opts=None):
+    """Load template variables from Rose suite configuration.
+
+    This loads the Rose suite configuration tree from the filesystem
+    using the shell environment.
+
+    Args:
+        dir_(string | pathlib.Path):
+            Path to the Rose suite configuration
+            (the directory containing the ``rose-suite.conf`` file).
+        opts:
+            Some sort of options object or string - To be used to allow CLI
+            specification of optional configuaration.
+
+    Returns:
+        dict - A dictionary of sections of rose-suite.conf.
+        For each section either a dictionary or None is returned.
+        E.g.
+            {
+                'env': {'MYVAR': 42},
+                'empy:suite.rc': None,
+                'jinja2:suite.rc': {
+                    'myJinja2Var': {'yes': 'it is a dictionary!'}
+                }
+            }
+
+    TODO:
+        - Consider allowing ``[jinja2:flow.conf]`` as an alias for
+          consistency with cylc.
```

Comment:
Convert to issue?

Reply:
https://github.com/cylc/cylc-rose/issues/25